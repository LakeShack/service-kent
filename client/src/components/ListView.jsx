import React from 'react';
import style from './styles/list-view.css';

var TourView = (props) => {

  const roomArr = [['Living area', 'living_room'], ['Dining area', 'dining_room'], ['Bedroom area', 'bedroom'], ['Den', 'den']];

  const renderRooms = roomArr.map((room) => {
    return (<div key={room[0]}>
      <img src={props.images[room[1]]} className={style.picture} />
      <p className={style.room_text}>{room[0]}</p>
    </div>);
  });


  return (
    <div className={style.list_view}>
      <div className={style.content}>
        {renderRooms}
      </div>
    </div>
  );
};

export default TourView;