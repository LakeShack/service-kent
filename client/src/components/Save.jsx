import React from 'react';
import style from './styles/picture-share-save-tour.css';

var Save = (props) => {

  const emptyHeart = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAANHSURBVGhD7ZlJyE1RHMA/8xwW5iHFxlBsEEKJlZBhRQiJQsnGmAUlsjJsFBKSIiEyrChjpEjGIgsLypx5/P2O79Tnq89373v3fe8t7q9+vXdO75x7zr33TP9XlZOTk5OTU0YaY+u/X4uiTfVng9EBF+JJfI7f8Te+wlu4Efvj/2iGk/AwPsLPaB0f8R7uwHHYCDOnBa7EN+hFo9/wNf6skafHsAfWZjI+w5q//YRva+XpVRyNmdENr2C8wCmcjeZHvMvDcDv6dPzde7Th0hz3Y6zjJi7GXhjxZg3GtRg76w1ahUU/nS4YK32II7E+2uIetIxPzE6fqE7byRlYHy1xNcZX1xtUMFZ2Da3oAnbENMzHX2h5fYw1n0ASxqNjx/KLzCgEx4QV+CTam1EA69A6HAP1TQJ1MR29IXao5uucCO9+HNijzCgQ3+2zaGOK4SDall0hlQJfCwueDqniaFf9WQx90IH/AZ0UEuM6YUfmhFRlcAlt08SQSshTtFDPkKoMNqBtWhNSCfmCPsqmIVUZOGvZkZ0hlQAbbwE7U0n4mtsu16jEvEMLZbEhzIrlaJu2hlRC7qOFBoZUZbANbdOKkErIIbTQvJCqDK6jbRobUglZihY6ElLlpzO67/qK7uUS0x2dtTwreAYpN3F8HA+plJxBC7tfKiceEeK6NtWMtIxBC3tw6mRGmViCtuMuerQuiPNoJW7ayoE7i3h6nGZGofRFj6JWNMuMBqQJXkSv7cGsaOKj9Tww1IwGwiCE132Bqc8hdXEArdTxMsSMErMJvZ5TrmM1MzwDeECy8pc4CEvFevQ6ngrnmpE1rfAcxsc9ArPEGWkzxk64dpQMN5FxJnN3bIQkC1ytY6TFhbjgQEMajK7sQy+qW7CYc0s/vIPWZSxsCjYoPvofaAMuY29My0yMRwbDRQOwLExAB78NMeqSNFriq7QXLacGOdLGzTKnK8Z9me7G/+1QXYuMlflbN6XLsCQB60KwITYoRtaNstdePJ2VDIMaRvU3t7GSDm7/4Dt+A22oDTbiYQccP4ZdzXdcOc2mik+VA2cwOxDvvH8PuCPw+wPMev0pOcPxCdoB9W+FVKe7SsLT5VFcEFI5OTk5OYGqqj9XWeCu2RXOqAAAAABJRU5ErkJggg==';
  const filledHeart = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAIISURBVGhD7dm/S1VhHMfx2w/sB5i5WQ0F4tjQmiKETTaJQRTkEKg0OvVjqU2aBG1qEXQIbSgV7A+oqLWhCESzITdNsUJLq/eHPHCRL9fnnHvueR7hecMLRfR7nqvnnnueaykWi8ViMY8dxPH/n1ZV/c7HwmpEL2bwFVv4ixW8xyAuoFLHcBUTWMAGNOMHPmIEHTiA3DuKu/gGHXQvL9GC8rSwm1iC9TO7vUU7cusUNNQ6WCWb6IHS6TcL6/sq2cY9VP3XacIXWAdx8Qd38Krsa1nodMucTqd3sAb70I9M6TdpDfRFF4PTSJWuTq5P7CI9QapuwRrk2zqOwLlpWINCcAXOfYY1JAT34Vzyahuix3DqMKwBoRiFc7rUWUNCMATnPsEaEoIBOPcM1pAQXIZzt2EN8e0ntA1w7gx052kN8+kFUqfNkzXMp1QvhkltsIb5oh2ottaZmoI11IdOZO4sdKNmDS6S9vdVdw3a6VkHKMIctK3IpUewDlJr33EeuXUIRT9f9FZTF3KvDkVdkn9Dp3TN0oN5DuvgedHbSN2oebrNH4a1iGot4xIKrQ+/YC0oiw9ohpcuYh7WwtIYwwl4TQsYh7XAvaziOoLqBtZgLdjyGucQZFrYG1gLT+jS+gC6aASdFvgQyf9Oyi2iFfsqbQO08ORBPEUD9mUnMQm9BRuLxWKxnUqlfyGi206x1fChAAAAAElFTkSuQmCC';
  let currentHeart = '';

  if (props.home.saved) {
    currentHeart = filledHeart;
  } else {
    currentHeart = emptyHeart;
  }

  return (
    <div>
      <button className={style.save_button} onClick={() => props.handlesaveclick()}>
        <img className={style.save_image} src={currentHeart} />
      </button>
    </div>
  );

};

export default Save;