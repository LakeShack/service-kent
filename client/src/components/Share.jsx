import React from 'react';
import style from './styles/picture-share-save-tour.css';
import { ToastContainer, toast } from 'react-toastify';
import '!style-loader!css-loader!react-toastify/dist/ReactToastify.css';

var Share = (props) => {

  const shareImage = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAALLSURBVHhe7Z3Bqw1hGIePBWFzUygWLOQvsFCUKDtlYytlSVnaKcpSNhZWlpQlKwr/gbKyQ8niriylpPi9t/fU900z58w3850zL+d56lnM1Nx+zVP3dufe5swAarNXPpJHdo5gUvbLt/KP/CSPSZgIi/FOWoy5nyVRJqAtBlEmYlGMuURZE31izCXKiimJMZcoK6Irxu/G8a/GsUmUynTF+CjvNs49kS8b50yiVGJRjMPyRnLOfCz3yLYoXyRRRrAshtEWxCBKZfrEMLqCGESpRN8YxqIgBlFGUhLDWBbEIMpASmMYfYIYRClkSAyjbxCDKD0ZGsMoCWIQZQm75GvZvEF9YhilQQyL8kKm15n2y+MBudGckc0b0zeGMSSI0RXlptxoTsj0eVRJDGNoEKMtyhW58VyT7+VTechOFDAmiGFRHsgP8r60b6EwgrFBoDIECQZBgkGQYBAkGAQJBkGCQZBgEMSx/zi/WNELcsiDvRpB9smzsm3XUA/KtWKPqtMbUcMf8rwsYWyQ4/KbTL9GDS/JtbKKIOZzWcLYILdlen0t/5sg92QJY4Nclun1tZw8yE/5ZoSv5B25W5ZQ42fIdWl/RWzb1ddtme6YPMhXOQU1gtTgmUx3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfMfGBrkq0x0P5RQQxDkqv0vbYO9MOSengCAJJ+UteXrnaBoIEgyCBIMgwSBIMAgSDIIEgyDBIEgwCBIMggSDIMEgSDAIEgyCBIMgwSBIMAgSDIIEgyDBIEgwCBIMggSDIMEgSDAIEgyCBIMgwSBIMAgSjHBBxr77/V833LvfMZcgwSRIMNcexD5l+RR2uiUHMJv9BT3oPXjaCC5CAAAAAElFTkSuQmCC';

  let shared;

  if (props.home.shared) {
    shared = 'This home has already been shared!';
  } else {
    shared = 'Successfully shared to all your social media!';
  }

  var notify = () => {
    toast(`${shared}`, {
      position: 'bottom-center',
      autoClose: 3000,
      hideProgressBar: true,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true
    });
  };

  return (
    <div>
      <button className={style.share_button} onClick={() => { props.handleshareclick(); notify(); }}><img className={style.share_image} src={shareImage} /></button>
      <ToastContainer
        position="bottom-center"
        autoClose={3000}
        hideProgressBar
        newestOnTop
        closeOnClick
        rtl={false}
        pauseOnVisibilityChange
        draggable
        pauseOnHover
      />
    </div>

  );

};

export default Share;