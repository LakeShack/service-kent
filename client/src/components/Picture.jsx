import React from 'react';
import Save from './Save.jsx';
import Share from './Share.jsx';
import Tour from './Tour.jsx';
import style from './styles/picture-share-save-tour.css';


var Picture = (props) => (

  <div className={style.container} >
    <span><img src={props.home.image.living_room} className={style.picture}/></span>
    <Save home={props.home} handlesaveclick={props.handlesaveclick}/>
    <Share home={props.home} handleshareclick={props.handleshareclick}/>
    <Tour home={props.home} handlepageviewclick={props.handlepageviewclick} tourview={props.tourview}/>   
  </div>
  
);

export default Picture;