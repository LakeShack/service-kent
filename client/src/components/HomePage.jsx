import React from 'react';
import axios from 'axios';
import Introduction from './Introduction.jsx';
import Picture from './Picture.jsx';
import TourPage from './TourPage.jsx';
import ListView from './ListView.jsx';
import queryString from 'query-string';
import { CSSTransitionGroup } from 'react-transition-group';


class HomePage extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      home: {},
      tourView: false,
      listView: false,
      currentId: 1
    };

    this.getImage = this.getImage.bind(this);
    this.handlePageViewClick = this.handlePageViewClick.bind(this);
    this.handleListViewClick = this.handleListViewClick.bind(this);
    this.saveHomeToList = this.saveHomeToList.bind(this);
    this.handleSaveClick = this.handleSaveClick.bind(this);
    this.shareHome = this.shareHome.bind(this);
    this.handleShareClick = this.handleShareClick.bind(this);
  }

  componentDidMount() {
    let parsed = queryString.parse(location.search);
    let currentId = Number(parsed.id);
    if (currentId) {
      this.getImage(Number(parsed.id));
      this.setState({ currentId: currentId });
    } else {
      this.getImage(1);
    }
  }

  getImage(id) {
    axios.get(`/images/${id}`)
      .then((response) => {
        this.setState({ home: response.data });
      })
      .catch(function (error) {
        return;
      });
  }

  saveHomeToList(id) {
    axios.patch(`/images/${id}`)
      .then((response) => {
        this.setState({ home: response.data });
      })
      .catch((error) => {
        console.log('CLIENT PATCH FAILED');
        return;
      });
  }

  shareHome(id) {
    axios.patch(`/images/${id}`, { 'shared': true })
      .then((response) => {
        this.setState({ home: response.data });
      })
      .catch((error) => {
        console.log('CLIENT PATCH FAILED');
        return;
      });
  }

  handlePageViewClick() {
    this.setState((prevState) => {
      return { tourView: !prevState.tourView };
    });
  }


  handleListViewClick() {
    this.setState((prevState) => {
      return { listView: !prevState.listView };
    });
  }

  handleSaveClick() {
    this.saveHomeToList(this.state.currentId);
  }

  handleShareClick() {
    this.shareHome(this.state.currentId);
    if (this.state.home.shared) {
      console.log('already shared');
    } else {
      console.log('shared');
    }
  }

  render() {

    return (
      <div className='home-page'>
        <Introduction home={this.state.home} />
        {this.state.home.image &&
          <Picture
            home={this.state.home}
            handlepageviewclick={this.handlePageViewClick}
            tourview={this.state.tourView}
            handlesaveclick={this.handleSaveClick}
            handleshareclick={this.handleShareClick} />
        }
        <div>
          <CSSTransitionGroup
            transitionName="tour-view"
            transitionEnterTimeout={600}
            transitionLeave={false}>
            {this.state.home.image && this.state.tourView ?
              <TourPage
                images={this.state.home.image}
                descriptions={this.state.home.descriptions}
                handlepageviewclick={this.handlePageViewClick}
                handlelistviewclick={this.handleListViewClick} />
              : null}
          </CSSTransitionGroup>
        </div>

        {this.state.home.image && this.state.tourView && this.state.listView ?
          <ListView images={this.state.home.image} />
          : null}


      </div>
    );
  }

}

export default HomePage;