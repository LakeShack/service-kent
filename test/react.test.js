import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import axios from 'axios';
import HomePage from '../client/src/components/HomePage.jsx';
import Introduction from '../client/src/components/Introduction.jsx';
import Picture from '../client/src/components/Picture.jsx';
import Save from '../client/src/components/Save.jsx';
import Share from '../client/src/components/Share.jsx';
import Tour from '../client/src/components/Tour.jsx';
import TourPage from '../client/src/components/TourPage.jsx';
import ListView from '../client/src/components/ListView.jsx';


const testData = {
  images: {
    living_room: 'living_room',
    bedroom: 'bedroom',
    dining_room: 'dining_room',
    den: 'den'
  },
  descriptions: {
    living_room: 'living_room description',
    bedroom: 'bedroom description',
    dining_room: 'dining_room description',
    den: 'den description'
  }
};

describe('HomePage', () => {

  const homePage = shallow(<HomePage />);

  it('HomePage component should render the home-page div', () => {
    expect(homePage.is('.home-page')).toBe(true);
  });

  it('should mount only 1', function () {
    expect(mount(<HomePage />).find('.home-page').length).toBe(1);
  });

  const wrapper = homePage;
  wrapper.state().tourView = false;
  wrapper.state().listView = false;

  it('handlePageViewClick in HomePage toggles the boolean in state.tourView', () => {

    expect(wrapper.state().tourView).toEqual(false);
    wrapper.instance().handlePageViewClick();
    expect(wrapper.state('tourView')).toEqual(true);
    wrapper.instance().handlePageViewClick();

  });

  it('handleListViewClick in HomePage toggles the boolean in state.listView', () => {

    expect(wrapper.state().listView).toEqual(false);
    wrapper.instance().handleListViewClick();
    expect(wrapper.state('listView')).toEqual(true);
    wrapper.instance().handleListViewClick();

  });

  const tourButton = shallow(<Tour handlepageviewclick={() => wrapper.setState({ tourView: true })} />);

  it('handlepageclickview in Tour toggles the boolean in state.tourView', () => {

    expect(wrapper.state().tourView).toEqual(false);
    tourButton.find('.tour_button').simulate('click');
    expect(wrapper.state('tourView')).toEqual(true);
    tourButton.find('.tour_button').simulate('click');

  });

  const tourButtons = shallow(<TourPage handlelistviewclick={() => wrapper.setState({ listView: true })} handlepageviewclick={() => wrapper.setState({ tourView: false })} images={testData.images} descriptions={testData.descriptions} />);

  it('handlepageclickview in TourPage toggles the boolean in state.tourView', () => {

    expect(wrapper.state().tourView).toEqual(true);
    tourButtons.find('.back_button').simulate('click');
    expect(wrapper.state('tourView')).toEqual(false);
    tourButtons.find('.back_button').simulate('click');

  });

  it('handlelistclickview in TourPage toggles the boolean in state.listView', () => {

    expect(wrapper.state().listView).toEqual(false);
    tourButtons.find('.tour_button').simulate('click');
    expect(wrapper.state('listView')).toEqual(true);
    tourButtons.find('.tour_button').simulate('click');

  });

  jest.mock('axios', () => {
    const sampleHome = [
      { title: 'test home', url: 'sample_url' }
    ];

    return {
      get: jest.fn(() => Promise.resolve(sampleHome)),
    };
  });

  it('get home on #componentDidMount', () => {
    const home = homePage;

    home.update();
    setTimeout(() => {
      expect(axios.get).toHaveBeenCalled();
      expect(axios.get).toHaveBeenCalledWith('sample_url');
      expect(app.state()).toHaveProperty('home', [
        { title: 'test home', url: 'sample_url' }
      ]);
    }, 1000);
  });
});

describe('Introduction', () => {

  const home = {
    location: 'test',
    rating: 3,
    title: 'test'
  };

  it('Introduction component should render the introduction div', () => {
    expect(shallow(<Introduction home={home}/>).is('.introduction')).toBe(true);
  });

  it('should mount only 1', function () {
    expect(mount(<Introduction home={home}/>).find('.introduction').length).toBe(1);
  });

});


describe('Picture', () => {

  it('should match snapshot', () => {
    const home = {
      image: {
        living_room: 'test'
      }
    };
    const view = renderer.create(
      <Picture home={home} />
    ).toJSON();

    expect(view).toMatchSnapshot();
  });

});

describe('Save', () => {

  it('should match snapshot', () => {
    const home = {
      saved: true
    };
    const view = renderer.create(
      <Save home={home} />
    ).toJSON();

    expect(view).toMatchSnapshot();
  });

});

describe('Share', () => {

  const home = {
    shared: false
  };

  it('should match snapshot', () => {
    const view = renderer.create(
      <Share home={home}/>
    ).toJSON();

    expect(view).toMatchSnapshot();
  });


});

describe('Tour', () => {

  it('should match snapshot', () => {
    const view = renderer.create(
      <Tour />
    ).toJSON();

    expect(view).toMatchSnapshot();
  });

});

describe('TourPage', () => {

  it('should match snapshot', () => {
    const images = ['tour', 'test'];
    const descriptions = ['descriptions', 'test'];
    const view = renderer.create(
      <TourPage images={images} descriptions={descriptions} />
    ).toJSON();

    expect(view).toMatchSnapshot();
  });

});

describe('ListView', () => {

  it('should match snapshot', () => {
    const images = ['list', 'test'];
    const view = renderer.create(
      <ListView images={images} />
    ).toJSON();

    expect(view).toMatchSnapshot();
  });

});