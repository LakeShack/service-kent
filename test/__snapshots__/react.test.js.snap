// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView should match snapshot 1`] = `
<div
  className="list_view"
>
  <div
    className="content"
  >
    <div>
      <img
        className="picture"
        src={undefined}
      />
      <p
        className="room_text"
      >
        Living area
      </p>
    </div>
    <div>
      <img
        className="picture"
        src={undefined}
      />
      <p
        className="room_text"
      >
        Dining area
      </p>
    </div>
    <div>
      <img
        className="picture"
        src={undefined}
      />
      <p
        className="room_text"
      >
        Bedroom area
      </p>
    </div>
    <div>
      <img
        className="picture"
        src={undefined}
      />
      <p
        className="room_text"
      >
        Den
      </p>
    </div>
  </div>
</div>
`;

exports[`Picture should match snapshot 1`] = `
<div
  className="container"
>
  <span>
    <img
      className="picture"
      src="test"
    />
  </span>
  <div>
    <button
      className="save_button"
      onClick={[Function]}
    >
      <img
        className="save_image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAANHSURBVGhD7ZlJyE1RHMA/8xwW5iHFxlBsEEKJlZBhRQiJQsnGmAUlsjJsFBKSIiEyrChjpEjGIgsLypx5/P2O79Tnq89373v3fe8t7q9+vXdO75x7zr33TP9XlZOTk5OTU0YaY+u/X4uiTfVng9EBF+JJfI7f8Te+wlu4Efvj/2iGk/AwPsLPaB0f8R7uwHHYCDOnBa7EN+hFo9/wNf6skafHsAfWZjI+w5q//YRva+XpVRyNmdENr2C8wCmcjeZHvMvDcDv6dPzde7Th0hz3Y6zjJi7GXhjxZg3GtRg76w1ahUU/nS4YK32II7E+2uIetIxPzE6fqE7byRlYHy1xNcZX1xtUMFZ2Da3oAnbENMzHX2h5fYw1n0ASxqNjx/KLzCgEx4QV+CTam1EA69A6HAP1TQJ1MR29IXao5uucCO9+HNijzCgQ3+2zaGOK4SDall0hlQJfCwueDqniaFf9WQx90IH/AZ0UEuM6YUfmhFRlcAlt08SQSshTtFDPkKoMNqBtWhNSCfmCPsqmIVUZOGvZkZ0hlQAbbwE7U0n4mtsu16jEvEMLZbEhzIrlaJu2hlRC7qOFBoZUZbANbdOKkErIIbTQvJCqDK6jbRobUglZihY6ElLlpzO67/qK7uUS0x2dtTwreAYpN3F8HA+plJxBC7tfKiceEeK6NtWMtIxBC3tw6mRGmViCtuMuerQuiPNoJW7ayoE7i3h6nGZGofRFj6JWNMuMBqQJXkSv7cGsaOKj9Tww1IwGwiCE132Bqc8hdXEArdTxMsSMErMJvZ5TrmM1MzwDeECy8pc4CEvFevQ6ngrnmpE1rfAcxsc9ArPEGWkzxk64dpQMN5FxJnN3bIQkC1ytY6TFhbjgQEMajK7sQy+qW7CYc0s/vIPWZSxsCjYoPvofaAMuY29My0yMRwbDRQOwLExAB78NMeqSNFriq7QXLacGOdLGzTKnK8Z9me7G/+1QXYuMlflbN6XLsCQB60KwITYoRtaNstdePJ2VDIMaRvU3t7GSDm7/4Dt+A22oDTbiYQccP4ZdzXdcOc2mik+VA2cwOxDvvH8PuCPw+wPMev0pOcPxCdoB9W+FVKe7SsLT5VFcEFI5OTk5OYGqqj9XWeCu2RXOqAAAAABJRU5ErkJggg=="
      />
    </button>
  </div>
  <div>
    <button
      className="share_button"
      onClick={[Function]}
    >
      <img
        className="share_image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAALLSURBVHhe7Z3Bqw1hGIePBWFzUygWLOQvsFCUKDtlYytlSVnaKcpSNhZWlpQlKwr/gbKyQ8niriylpPi9t/fU900z58w3850zL+d56lnM1Nx+zVP3dufe5swAarNXPpJHdo5gUvbLt/KP/CSPSZgIi/FOWoy5nyVRJqAtBlEmYlGMuURZE31izCXKiimJMZcoK6Irxu/G8a/GsUmUynTF+CjvNs49kS8b50yiVGJRjMPyRnLOfCz3yLYoXyRRRrAshtEWxCBKZfrEMLqCGESpRN8YxqIgBlFGUhLDWBbEIMpASmMYfYIYRClkSAyjbxCDKD0ZGsMoCWIQZQm75GvZvEF9YhilQQyL8kKm15n2y+MBudGckc0b0zeGMSSI0RXlptxoTsj0eVRJDGNoEKMtyhW58VyT7+VTechOFDAmiGFRHsgP8r60b6EwgrFBoDIECQZBgkGQYBAkGAQJBkGCQZBgEMSx/zi/WNELcsiDvRpB9smzsm3XUA/KtWKPqtMbUcMf8rwsYWyQ4/KbTL9GDS/JtbKKIOZzWcLYILdlen0t/5sg92QJY4Nclun1tZw8yE/5ZoSv5B25W5ZQ42fIdWl/RWzb1ddtme6YPMhXOQU1gtTgmUx3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfMfGBrkq0x0P5RQQxDkqv0vbYO9MOSengCAJJ+UteXrnaBoIEgyCBIMgwSBIMAgSDIIEgyDBIEgwCBIMggSDIMEgSDAIEgyCBIMgwSBIMAgSDIIEgyDBIEgwCBIMggSDIMEgSDAIEgyCBIMgwSBIMAgSjHBBxr77/V833LvfMZcgwSRIMNcexD5l+RR2uiUHMJv9BT3oPXjaCC5CAAAAAElFTkSuQmCC"
      />
    </button>
    <div />
  </div>
  <div>
    <div>
       
      <button
        className="tour_button"
        onClick={[Function]}
      >
        <img
          className="tour_image"
          src="https://image.flaticon.com/icons/svg/54/54917.svg"
        />
        TOUR THIS HOME
      </button>
       
    </div>
  </div>
</div>
`;

exports[`Save should match snapshot 1`] = `
<div>
  <button
    className="save_button"
    onClick={[Function]}
  >
    <img
      className="save_image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAIISURBVGhD7dm/S1VhHMfx2w/sB5i5WQ0F4tjQmiKETTaJQRTkEKg0OvVjqU2aBG1qEXQIbSgV7A+oqLWhCESzITdNsUJLq/eHPHCRL9fnnHvueR7hecMLRfR7nqvnnnueaykWi8ViMY8dxPH/n1ZV/c7HwmpEL2bwFVv4ixW8xyAuoFLHcBUTWMAGNOMHPmIEHTiA3DuKu/gGHXQvL9GC8rSwm1iC9TO7vUU7cusUNNQ6WCWb6IHS6TcL6/sq2cY9VP3XacIXWAdx8Qd38Krsa1nodMucTqd3sAb70I9M6TdpDfRFF4PTSJWuTq5P7CI9QapuwRrk2zqOwLlpWINCcAXOfYY1JAT34Vzyahuix3DqMKwBoRiFc7rUWUNCMATnPsEaEoIBOPcM1pAQXIZzt2EN8e0ntA1w7gx052kN8+kFUqfNkzXMp1QvhkltsIb5oh2ottaZmoI11IdOZO4sdKNmDS6S9vdVdw3a6VkHKMIctK3IpUewDlJr33EeuXUIRT9f9FZTF3KvDkVdkn9Dp3TN0oN5DuvgedHbSN2oebrNH4a1iGot4xIKrQ+/YC0oiw9ohpcuYh7WwtIYwwl4TQsYh7XAvaziOoLqBtZgLdjyGucQZFrYG1gLT+jS+gC6aASdFvgQyf9Oyi2iFfsqbQO08ORBPEUD9mUnMQm9BRuLxWKxnUqlfyGi206x1fChAAAAAElFTkSuQmCC"
    />
  </button>
</div>
`;

exports[`Share should match snapshot 1`] = `
<div>
  <button
    className="share_button"
    onClick={[Function]}
  >
    <img
      className="share_image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAALLSURBVHhe7Z3Bqw1hGIePBWFzUygWLOQvsFCUKDtlYytlSVnaKcpSNhZWlpQlKwr/gbKyQ8niriylpPi9t/fU900z58w3850zL+d56lnM1Nx+zVP3dufe5swAarNXPpJHdo5gUvbLt/KP/CSPSZgIi/FOWoy5nyVRJqAtBlEmYlGMuURZE31izCXKiimJMZcoK6Irxu/G8a/GsUmUynTF+CjvNs49kS8b50yiVGJRjMPyRnLOfCz3yLYoXyRRRrAshtEWxCBKZfrEMLqCGESpRN8YxqIgBlFGUhLDWBbEIMpASmMYfYIYRClkSAyjbxCDKD0ZGsMoCWIQZQm75GvZvEF9YhilQQyL8kKm15n2y+MBudGckc0b0zeGMSSI0RXlptxoTsj0eVRJDGNoEKMtyhW58VyT7+VTechOFDAmiGFRHsgP8r60b6EwgrFBoDIECQZBgkGQYBAkGAQJBkGCQZBgEMSx/zi/WNELcsiDvRpB9smzsm3XUA/KtWKPqtMbUcMf8rwsYWyQ4/KbTL9GDS/JtbKKIOZzWcLYILdlen0t/5sg92QJY4Nclun1tZw8yE/5ZoSv5B25W5ZQ42fIdWl/RWzb1ddtme6YPMhXOQU1gtTgmUx3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfAdBXIK4BMl3EMQliEuQfMfGBrkq0x0P5RQQxDkqv0vbYO9MOSengCAJJ+UteXrnaBoIEgyCBIMgwSBIMAgSDIIEgyDBIEgwCBIMggSDIMEgSDAIEgyCBIMgwSBIMAgSDIIEgyDBIEgwCBIMggSDIMEgSDAIEgyCBIMgwSBIMAgSjHBBxr77/V833LvfMZcgwSRIMNcexD5l+RR2uiUHMJv9BT3oPXjaCC5CAAAAAElFTkSuQmCC"
    />
  </button>
  <div />
</div>
`;

exports[`Tour should match snapshot 1`] = `
<div>
  <div>
     
    <button
      className="tour_button"
      onClick={[Function]}
    >
      <img
        className="tour_image"
        src="https://image.flaticon.com/icons/svg/54/54917.svg"
      />
      TOUR THIS HOME
    </button>
     
  </div>
</div>
`;

exports[`TourPage should match snapshot 1`] = `
<div
  className="tour_page"
>
  <div
    className="navigation"
  >
    <button
      className="back_button"
      onClick={[Function]}
    >
      <img
        className="back_image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAG5SURBVHhe7d0xSiRBAEbhQVcNnFQFc0HwFCKY6AU2MNpkQ2P1FkamXkHQPcGeQfNNBA+ggat/wQjSFDodVPmQ98FP5fXotGsiSZL0nU1np77Qj+w0+5e9ZPfZWbaYqbOV7DorIYa7yNRRiXGT1WKU/c92MnXwWYy3/crU2Lwxyo4yNTQmxmO2nqmRMTHKfmdqZGyMk0yNGAPEGCDGADEGiDFAjAFiDBBjgBgDxBggxgAxBogxQIwBYgwQY4AYA8QYIMYAMQaIMUCMAWIMEGOAGAPEGCDGADEGiDFAjAFiDBBjgCxnxgA5z2oXX5sxGlvKnrLa5Q9njA62s9rlD3eZqYPV7DmrRXi/8hUdZurgKqtFGM4onaxld1ktwnBG6WQzMwqMUYCMAmQUIKMAjY1ykKkxowAZBcgoQEYBMgqQUYCMAmQUIKMAGQXIKEBGATIKkFGAjAJUotxmtQjDGaUTowAZBcgoQEYBMgqQUYCMAmQUIKMAGQXIKEBjo+xlamxMlIdsmqmxMVF+Zupg3ijHmTqZJ8pupo4+ivI3W8jUWXnE/k/2Pkb5e9FGpi+0le3PTkmSJH0Dk8kr2Milf2sdpAkAAAAASUVORK5CYII="
      />
    </button>
    <button
      className="tour_button"
      onClick={[Function]}
    >
      <img
        className="tour_image"
        src="https://image.flaticon.com/icons/svg/54/54917.svg"
      />
      Tour this home
    </button>
  </div>
  <div
    className="content top"
  >
    <div>
      <p
        className="room_text"
      >
        Living area
      </p>
      <p
        className="description_text"
      />
      <img
        className="picture"
        src={undefined}
      />
    </div>
    <div>
      <p
        className="room_text"
      >
        Dining area
      </p>
      <p
        className="description_text"
      />
      <img
        className="picture"
        src={undefined}
      />
    </div>
    <div>
      <p
        className="room_text"
      >
        Bedroom area
      </p>
      <p
        className="description_text"
      />
      <img
        className="picture"
        src={undefined}
      />
    </div>
    <div>
      <p
        className="room_text"
      >
        Den
      </p>
      <p
        className="description_text"
      />
      <img
        className="picture"
        src={undefined}
      />
    </div>
  </div>
</div>
`;
